src/main/java/com/minecraftabnormals/abnormals_core/core/api/model/FullbrightModelLoader.java

```diff
@@ -0,0 +1,259 @@
+        Set<RenderMaterial> textures = Sets.newHashSet();
+        for (FullbrightBlockPart part : this.elements) {
+            for (BlockPartFace face : part.mapFaces.values()) {
+                textures.add(owner.resolveTexture(face.texture));
+            }
+        }
+
+        return textures;
+    }
+
+    public static class Loader implements IModelLoader<FullbrightModelLoader> {
+        public static final FullbrightModelLoader.Loader INSTANCE = new FullbrightModelLoader.Loader();
+
+        private Loader() {}
+
+        @Override
+        public void onResourceManagerReload(IResourceManager manager) {}
+
+        @Override
+        public FullbrightModelLoader read(JsonDeserializationContext context, JsonObject model) {
+            List<FullbrightBlockPart> list = this.getModelElements(model);
+            return new FullbrightModelLoader(list);
+        }
+
+        private List<FullbrightBlockPart> getModelElements(JsonObject model) {
+            List<FullbrightBlockPart> list = Lists.newArrayList();
+            if (model.has("elements")) {
+                for (JsonElement jsonelement : JSONUtils.getJsonArray(model, "elements")) {
+                    list.add(GSON.fromJson(jsonelement, FullbrightBlockPart.class));
+                }
+            }
+
+            return list;
+        }
+    }
+
+    public static class FullbrightBlockPartFace extends BlockPartFace {
+        public boolean fullbright;
+        public boolean override;
+
+        public FullbrightBlockPartFace(@Nullable Direction cullFaceIn, int tintIndexIn, String textureIn, BlockFaceUV blockFaceUVIn, boolean fullbright, boolean override) {
+            super(cullFaceIn, tintIndexIn, textureIn, blockFaceUVIn);
+            this.fullbright = fullbright;
+            this.override = override;
+        }
+
+        public static class Deserializer extends BlockPartFace.Deserializer {
+            @Override
+            public BlockPartFace deserialize(JsonElement element, Type type, JsonDeserializationContext context) throws JsonParseException {
+                BlockPartFace face = super.deserialize(element, type, context);
+                JsonObject object = element.getAsJsonObject();
+
+                if (object.has("fullbright") && !JSONUtils.isBoolean(object, "fullbright"))
+                    throw new JsonParseException("Expected fullbright to be a Boolean");
+                return new FullbrightBlockPartFace(
+                        face.cullFace,
+                        face.tintIndex,
+                        face.texture,
+                        face.blockFaceUV,
+                        JSONUtils.getBoolean(object, "fullbright", false),
+                        object.has("fullbright") && JSONUtils.isBoolean(object, "fullbright")
+                );
+            }
+        }
+    }
+
+    public static class FullbrightBlockPart extends BlockPart {
```

Comment:
Do you intend for this class to be used outside of `FullbrightModelLoader`?
If not, then the access modifiers inside this class could all be private.